core/dnsserver/server.go

```diff
@@ -314,9 +320,32 @@ func DefaultErrorFunc(w dns.ResponseWriter, r *dns.Msg, rc int) {
 	w.WriteMsg(answer)
 }
 
+// depthCheck increments the loop counter in the context, and returns an error if
+// the counter exceeds the max number of re-entries
+func depthCheck(ctx context.Context) (context.Context, error) {
```

Comment:
incrementDepthAndCheck (or something better - depthCheck doesn't convey it is also incrementing)